pyflakes/test/test_imports.py

```diff
@@ -564,6 +563,15 @@ def bar():
         fu
         ''', m.UnusedImport, m.UndefinedName)
 
+    def test_shadowedInNestedScope(self):
+        self.flakes('''
+        import fu
+        def bar():
+            import fu
+            fu
+        fu
+        ''', m.ImportShadowedByImport)
```

Comment:
But this isn't even bad style. It's no different from

```
var = 1

def func():
    var = 2
    ...
```

Redefining variables in inner scopes is a perfectly legitimate thing to do, whether that be via imports, loops, `as` constructs, or assignments. Python's scoping rules mostly protect you from this doing anything bad.